getting-started: Align with upstream getting started snippets, comment about pedagogical code, and reduce stack usage

Patater opened the pull request:
- Improve the example by commenting where global keys are used for strictly pedagogical purposes.
- Align code snippets with changes in Mbed Crypto's getting started guide, to ensure we keep testing our getting started guide (as this example is the only time those snippets are tested, currently)
- Reduce stack usage by using global `static const` keys, instead of `#defines` for initializer lists to place keys on the stack.

Patater commented:
Tested with ARMC6 and the develop profile. IAR not yet tested.

gilles-peskine-arm commented on getting-started/main.cpp:
```diff
@@ -142,9 +142,9 @@ static void sign_a_message_using_rsa(void)
     psa_status_t status;
     psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
-    uint8_t key[] = RSA_KEY;
+    static uint8_t key[] = RSA_KEY;
```
Having a `static` variable is unusual, and usually done when the value needs to be preserved between calls, which is not the case here. Please add a comment in the code to indicate why these two variables are `static`.

gilles-peskine-arm replied:
Preexisting, but would be nice to do what you're at it: `key` should be `const`.

gilles-peskine-arm reviewed (changes requested):
Ok for the workaround, but this should be documented as a workaround in the code itself, especially since this is in example code.

Patater commented:
This is not meant as any sort of workaround. This is meant to reduce stack usage to make the example more portable. We could use a global key instead if use of static in this situation appears uncouth.

Patater commented:
New patchset uses `static const` for `key` and adds a comment explaining why the writable `signature` is `static`, as that might be a bit surprising. `static const` is idiomatic enough that it doesn't need a comment.

Patater commented on getting-started/main.cpp:
```diff
@@ -142,9 +142,10 @@ static void sign_a_message_using_rsa(void)
     uint8_t hash[] = "INPUT_FOR_SIGN";
-    uint8_t signature[PSA_ASYMMETRIC_SIGNATURE_MAX_SIZE] = {0};
+    /* 'signature' is static to reduce stack usage. */
+    static uint8_t signature[PSA_ASYMMETRIC_SIGNATURE_MAX_SIZE] = {0};
```
Making `signature` static is not necessary for the targets we run this example on so far. So, if we dislike adding static here, we can remove it as we don't need it yet. Just making the key `static const` is good enough to pass the tests with ARMC6-develop at least.

Patater commented:
Buddy PR at https://github.com/ARMmbed/mbed-crypto/pull/317

dgreen-arm reviewed (approved):
With these changes, the example works with latest Mbed OS using ARMC6 or IAR

Patater committed:
getting-started: Explain why we have global keys

Make it more clear that one should not have global, hard-coded keys in
their code. We do this for example purposes only and real-world
applications should follow secure practices.

Patater committed:
getting-started: Align with getting_started.md snippets

The getting started guide in Mbed Crypto was recently changed to make it
more obvious that `AES_KEY` and `RSA_KEY` were shorthand for key
material. The guide did this by using function parameters instead of
all-caps shorthand. In order to better match what is in our getting
started guide with this getting started example, we make the same
function parameter changes here, passing global key data in via
`main()`.

Passing global key data in via `main()` has the additional benefit of
reducing stack usage of each example snippet function. This enables the
example to run on more boards, those with more limited stack space.

Patater commented:
New patchset makes the keys global `static const` instead to reduce RAM usage, while also improving the readability of the example.

dgreen-arm reviewed (approved):
Example runs successfully on ARMC6 and IAR, LGTM

gilles-peskine-arm commented:
e46b6adcefcfe0eff770f682f9ca513092cd9751 passed CI with [ARM Compiler reinstated](https://github.com/ARMmbed/mbedtls-test/pull/116) in https://jenkins-internal.mbed.com/job/mbed-crypto-release-ci-testing/577 .
